### ISSUE TITLE
Update issue removing assignee

### ISSUE THREAD

**Issue by alenkis** [ahungry/org-jira#296]
When an `Unassigned` is used for assignee

```
:assignee: Unassigned
```
when running `org-jira-update-issue`, the issue is updated correctly but after the update (and when refreshing), the field is reset to:

```
:assignee:
```

On the successive updates, the issue will select the first user alphabetically. 

This is most likely related to https://github.com/ahungry/org-jira/issues/242

**Comment by xorb0ss** [comment]
Getting the same issue

**Comment by adanylkov** [comment]
+1

**Comment by adanylkov** [comment]
in put payload instead of actual ids there are nulls

![image_2023-02-28_16-18-12](https://user-images.githubusercontent.com/102509918/221897124-d3b119e6-2b3b-4521-85b0-1ca7ff5bca8c.png)

**Comment by ahungry** [comment]
Thanks for the reminder on this - the problem was on the read out of data, assignee would not have a value (name) mapping so on issue rendering, this would appear:

```
:assignee:
:filename: Whatever.org
```

Instead of this, which would be expected:

```
:assignee: Unassigned
:filename: Whatever.org
```

Which didn't make sense as "Unassigned" was already a special case on the input side (if set to this string, the assignee would be set to null on the remote).

@adanylkov what you're observing is the assignee in your screenshot was not found in the list of `(org-jira-get-assignable-users)`, triggering the input to resolve to a null (and essentially unassigning).

Try out something like `(org-jira-get-assignable-users "YOUR-PROJ-KEY")` in `M-x ielm` or something - see if the first/last combo you're attempting to set it to is found in that list or not.

I think some very large jiras have pagination issues with the assignee users list, which may require some customization of it on your behalf.

**Comment by adanylkov** [comment]
> 

It isn't working with real names too :(

this is output after `org-jira-update-issue`

![image_2023-02-28_17-38-26](https://user-images.githubusercontent.com/102509918/221918701-5acc97a6-5dfb-47fe-9a2e-012bc898704c.png)

**Comment by ahungry** [comment]
@adanylkov I had a weird issue when I started working on this today - when I ran `M-x eval-expression`, then entered:  `(org-jira-get-issue-val-from-org 'assignee)` - it was not giving me the assignee property (or any other property I tried it on).

Instead, I was receiving the "ID" value.  In my case, I saw it produce nested {"id":null} data pieces, however after re-loading my org-jira source code (open the file "org-jira.el" and run `M-x eval-buffer`) the error stopped.

I wonder if org itself was snap shotted in a way (with elc or eln files) that was causing this condition.

If you do the eval-buffer step, does your issue go away?

**Comment by adanylkov** [comment]
@ahungry, I ran M-x eval-expression, then (org-jira-get-issue-val-from-org 'assignee) and, as you said, got "BH-32". After running M-x eval-buffer on "org-jira.el", the error stopped. I was able to use the update-issue command, and all the fields in the put payload were filled. After running M-x eval-buffer, I got two messages: "Created jira link" and "Package cl is deprecated". 

However, this workaround is not persistent. After restarting emacs, I got the same "BH-32" error. If there is anything else I can do to help you track the problem, I would be happy to assist.
